cmd_watch.go

```diff
@@ -113,29 +117,32 @@ build with the appropriate timers.`,
 
 // waitCircle polls the CircleCI API checking for the status of this workflow
 // and the jobs it contains. It returns whether the workflow build succeeded,
-// the time it started. The err returned is for errors polling the CircleCI API,
-// not errors in the build itself.
-func waitCircle(parent context.Context, cfg watchConfig) (bool, time.Time, error) {
+// the time it started, and the time it ended (which will be either nowish or
+// sometime in the past if we timed out). The err returned is for errors polling
+// the CircleCI API, not errors in the build itself.
+func waitCircle(parent context.Context, cfg watchConfig) (passed bool, started, ended time.Time, err error) {
 	// we need a token to query anything; give a helpful error if we have no token
 	if cfg.circleKey == "" {
-		return false, time.Now(), fmt.Errorf("circle token required to poll the API")
+		return false, time.Now(), time.Now().Add(time.Second), fmt.Errorf("circle token required to poll the API")
```

Comment:
curious why we wouldn't just return zero times here and document that.

Reply:
that'd probably be fine too... I figured this would stay away from "is it unset or the default?" problems. There's definitely a better answer out here somewhere.